backend/src/apiserver/client/pod.go

```diff
@@ -0,0 +1,45 @@
+package client
+
+import (
+	"github.com/cenkalti/backoff"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"time"
+)
+
+func CreatePodClient(namespace string) (v1.PodInterface, error) {
+	restConfig, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to initialize kubernetes client.")
+	}
+
+	clientSet, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to initialize kubernetes client set.")
+	}
+	return clientSet.CoreV1().Pods(namespace), nil
+}
+
+// creates a new client for the Kubernetes pod.
+func CreatePodClientOrFatal(namespace string, initConnectionTimeout time.Duration) v1.PodInterface{
```

Comment:
Format the line?

Also, do we really want to die here? This kills the APIServer. Why not just return an error to the client? This is a blocking call anyway.

Reply:
this is called during api server startup. it implies something wrong with cluster that can't be recovered. this is what we do for other clients in the same folder too.